src/github.com/getlantern/peerscanner/main.go

```diff
@@ -69,122 +107,238 @@ func main() {
 
 func parseFlags() {
 	flag.Parse()
-	if cfuser == "" {
-		log.Fatal("Please specify a CF_USER environment variable")
+	if cflid == "" {
+		log.Fatal("Please specify a CFL_ID environment variable")
+	}
+	if cflkey == "" {
+		log.Fatal("Please specify a CFL_KEY environment variable")
+	}
+	if cfrid == "" {
+		log.Fatal("Please specify a CFR_ID environment variable")
 	}
-	if cfkey == "" {
-		log.Fatal("Please specify a CF_API_KEY environment variable")
+	if cfrkey == "" {
+		log.Fatal("Please specify a CFR_KEY environment variable")
+	}
+	if dspid == "" {
+		log.Fatal("Please specify a DSP_ID environment variable")
+	}
+	if dspkey == "" {
+		log.Fatal("Please specify a DSP_KEY environment variable")
 	}
 }
 
 func connectToCloudFlare() {
 	log.Debug("Connecting to CloudFlare ...")
-	cfutil = cf.New(*cfdomain, cfuser, cfkey)
+	cflutil = cfl.New(*cfldomain, cflid, cflkey)
+}
+
+func connectToCloudFront() {
+	log.Debug("Connecting to CloudFront ...")
+	cfrutil = cfr.New(cfrid, cfrkey, nil)
+}
+
+func connectToDnsimple() {
+	log.Debug("Connecting to DNSimple ...")
+	dsputil = dsp.New(*dspdomain, dspid, dspkey)
 }
 
 /*******************************************************************************
  * Functions for managing map of hosts
  ******************************************************************************/
 
-// loadHosts loads the initial list of hosts based on what's in CloudFlare's
-// DNS at startup.
+// loadHosts loads the initial list of hosts based on the existing entries in
+// the CDN and DNS services we manage
 func loadHosts() (map[string]*host, error) {
-	log.Debug("Loading existing hosts from CloudFlare ...")
 
-	recs, err := cfutil.GetAllRecords()
+	log.Debug("Loading existing CloudFlare records ...")
+	cflRecs, err := cflutil.GetAllRecords()
 	if err != nil {
-		return nil, fmt.Errorf("Unable to load hosts: %v", err)
+		return nil, fmt.Errorf("Unable to load Cloudflare records: %v", err)
 	}
+	log.Debugf("Loaded %d existing Cloudflare records", len(cflRecs))
 
-	log.Debugf("Loaded %d existing hosts", len(recs))
+	log.Debug("Loading existing DNSimple records ...")
+	dspRecs, err := dsputil.GetAllRecords()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to load DNSimple records: %v", err)
+	}
+	log.Debugf("Loaded %d existing DNSimple records", len(dspRecs))
 
-	// Keep track of different groups of hosts
-	groups := make(map[string]map[string]*cloudflare.Record, 0)
+	dists, err := cfr.ListDistributions(cfrutil)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to load cloudfront distributions: %v", err)
+	}
+	log.Debugf("Loaded %d existing distributions", len(dists))
 
-	addToGroup := func(name string, r cloudflare.Record) {
+	// Collect round-robin entries in Cloudflare
+	cflGroups := make(map[string]map[string]*cloudflare.Record, 0)
```

Comment:
It seems like there's an opportunity for refactoring here. Like what about a CloudfrontRecords and CloudFlareRecords struct, each of which implements an interface with add, remove, etc, and that just looks up all the records when it's created, something like that? Seems like that could simplify things a bit, but this certainly looks like it works for the purposes of getting this out the door.